ui/shared/components/panel/view-fields/TagFieldView.jsx

```diff
@@ -96,18 +96,22 @@ const TagFieldView = ({ initialValues, field, tagOptions, popup, tagAnnotation,
     tagOptions={tagOptions}
     formFieldProps={formFieldProps}
     additionalEditFields={additionalFields}
-    initialValues={mappedValues}
+    initialValues={initialValues}
     modalStyle={MODAL_STYLE}
     fieldDisplay={displayFieldValues =>
       <span>
         {displayFieldValues.map((tag) => {
           const label = <ColoredLabel size="small" color={tag.color} horizontal content={tag.name || tag.text} />
           return (
-            <span key={tag.name}>
-              <HorizontalSpacer width={5} />
-              {popup ? popup(tag)(label) : label}
-              {tagAnnotation && tagAnnotation(tag)}
-            </span>
+            showIconOnly ?
```

Comment:
Rather than adding all this  logic so you can display these as colored dots with text, just display them as colored labels the way this component defaults to. Its fine, even if its not whats shown in the ticket

Reply:
This ensures backward compatibility with FamilyVariantTag, cause all details such as color/categories etc were stored in the store for FamilyVariantTags, but for SuccessStoryType one Char is stored.

Reply:
But `FAMILY_SUCCESS_STORY_TYPE_OPTIONS` can include the correct mapping between the stored value and displayed name, so you should be able to use that. You may need to update them so that `name` is the actual stored one-char value and then add a `label` field with the text you want to show